### PR TITLE
Master phoenix fge

### PR DESCRIPTION
**Pull request by FrancoisGe**

### CODE REVIEW

File: addons/html_editor/static/src/editor/list/list_plugin.js [hunk 1]
```diff
@@ -323,7 +305,7 @@ export class ListPlugin extends Plugin {
             }
         }
 
-        const selectionToRestore = this.shared.getEditableSelection();
+        const selectionToRestore = { ...this.shared.getEditableSelection() };
```
Comment (ged-odoo) [inline addons/html_editor/static/src/editor/list/list_plugin.js:308]: why clone the selection?

File: addons/html_editor/static/src/editor/core/selection_plugin.js [hunk 2]
```diff
@@ -117,13 +114,11 @@ export class SelectionPlugin extends Plugin {
     }
 
     setCursorStart(node) {
-        const pos = startPos(node);
-        return this.setSelection(...pos, ...pos);
+        return this.setSelection({ anchorNode: node, anchorOffset: 0 });
```
Comment (ged-odoo) [inline addons/html_editor/static/src/editor/core/selection_plugin.js:117]: not sure if we want to normalize here. maybe in this case, it's useless?

File: addons/html_editor/static/src/editor/core/selection_plugin.js [hunk 3]
```diff
@@ -117,13 +114,11 @@ export class SelectionPlugin extends Plugin {
...
     }
 
     setCursorEnd(node) {
-        const pos = endPos(node);
-        return this.setSelection(...pos, ...pos);
+        return this.setSelection({ anchorNode: node, anchorOffset: nodeSize(node) });
```
Comment (ged-odoo) [inline addons/html_editor/static/src/editor/core/selection_plugin.js:121]: same here

File: addons/html_editor/static/src/editor/core/format_plugin.js [hunk 4]
```diff
@@ -245,15 +252,27 @@ export class FormatPlugin extends Plugin {
         }
 
         if (selectedTextNodes[0] && selectedTextNodes[0].textContent === "\u200B") {
-            this.shared.setSelection(selectedTextNodes[0], 0);
+            this.shared.setCursorStart(selectedTextNodes[0]);
         } else if (selectedTextNodes.length) {
             const firstNode = selectedTextNodes[0];
             const lastNode = selectedTextNodes[selectedTextNodes.length - 1];
+            let newSelection;
```
Comment (ged-odoo) [inline addons/html_editor/static/src/editor/core/format_plugin.js:259]: not very important, but i would have done this instead:
```js
let newSelection = direction === DIRECTIONS.RIGHT 
     ? { anchorNode: ... }
    : { anchorNode: ... };
this.shared.setSelection(newSelection, false);
```

File: addons/html_editor/static/src/editor/list/list_plugin.js (hunk 1 above)
Comment (cammarosano) [reply addons/html_editor/static/src/editor/list/list_plugin.js:308]: He changes some properties of this cloned object a few lines below, so he can call `this.shared.setSelection(selectionToRestore, false);` with it. (see lines 364 - 369)

**Pull request merged**